### PR TITLE
StructScanner to register custom type converters

### PR DESCRIPTION
**Pull request by lgonzalez-silen**

### CODE REVIEW

**Comment by lgonzalez-silen** [comment]
This impl is a bit flawed because dests inside StructScanner should not make dests available outside the context of a single StructScan run.

Worksheet.StructScan is ok because it creates its own every time, but if someone creates a StructScanner for reuse they will run into problems (putting the same ws.Id() into dests twice will panic; concurrent access is not safe).

File: marshaling.go [hunk 1]
```diff
@@ -131,31 +131,94 @@ type WorksheetConverter interface {
...
+		// dests are stored as pointers but we are setting non-pointer destinations
+		locus.Set(destPtr.Elem())
+	}
+}
+
+type wsDestinationMap map[string]*wsDestination
+
+func (wsdm wsDestinationMap) setAllDestinations() {
+	for _, d := range wsdm {
+		d.setAll()
```
Comment (pascallouisperez) [inline marshaling.go]: Consider inlining for clarity, since it's unlikely we'd use `setAll` by itself.

File: marshaling.go [hunk 2]
```diff
@@ -131,31 +131,94 @@ type WorksheetConverter interface {
...
+	}
+}
+
+func (wsdm wsDestinationMap) newDestination(destId string, dest interface{}) {
```
Comment (pascallouisperez) [inline marshaling.go]: rename to `addDestination`?

and signature could be `addDestination(ws *Worksheet, dest interface{})`

File: marshaling.go [hunk 3]
```diff
@@ -131,31 +131,94 @@ type WorksheetConverter interface {
...
+	if _, ok := wsdm[destId]; ok {
+		panic("incorrect usage: cannot add new destination multiple times")
+	}
+	wsdm[destId] = &wsDestination{dest, nil}
+}
+
+func (wsdm wsDestinationMap) addLocus(destId string, locus reflect.Value) {
```
Comment (pascallouisperez) [inline marshaling.go]: signature could be `addLocus(ws *Worksheet, locus reflect.Value)`

**Pull request merged**